R/tribble.R

```diff
@@ -19,6 +30,26 @@
 #'   "c",   3
 #' )
 #'
+#' # Conversion expressions can be supplied, including additional
+#' # paremeters if needed.
+#' tribble(
+#'   ~colA, colB~factor(., levels = c("B", "A")),
```

Comment:
Regarding the use of `=` rather than `~`, I'd been thinking along the same lines. One issue is that `list(...)` fails when using unquoted expressions because it tries to evaluate them. I'd need to do something along the line of `quos(...)`, but this ignores named arguments with no value such as `colA =`. Without chaning `rlang`, the options that I can see are to require the syntax `colA=.` for unconverted columns, and preserve the existing behaviour of supplying a `formula`. e.g.

    tribble(
      colA = ., colB = lubridate::ymd(.),
      1, "2013-01-02"
    )

equivalently:

    tribble(
      ~colA, colB = lubridate::ymd(.),
      1, "2013-01-02"
    )

With regards to `magrittr`, I actually started off by using the pipe from `magrittr`, then using an adpated version of the `magrittr` code, and then saw that your preferred syntax at the time was `~as.Date(colA)`. One issue with making the dot optional is what do to with syntax like this:

    tribble(colA~as.character(as.factor()), 1)

If passed to `magrittr` or interpreted using the same assumptions, this becomes:

    tribble(colA~as.character(., as.factor()), 1)

which is invalid syntax, and potentially confusing. We could either suggest that package users use pipes for more complex expressions, or could refuse to automatically add the placeholder once there is more than one nested call.